Content.Server/Doors/Systems/DoorSystem.cs

```diff
@@ -148,9 +150,15 @@ private void OnWeldAttempt(EntityUid uid, DoorComponent component, WeldableAttem
     private void OnWeldChanged(EntityUid uid, DoorComponent component, WeldableChangedEvent args)
     {
         if (component.State == DoorState.Closed)
+        {
             SetState(uid, DoorState.Welded, component);
+            AdjustLayer(uid, true);
+        }
         else if (component.State == DoorState.Welded)
+        {
             SetState(uid, DoorState.Closed, component);
+            AdjustLayer(uid, false);
```

Comment:
why is this changed in the door handler? `LayerChangeOnWeldComponent` should have its own subscription so it isnt tied to door. if this is needed because of some indirection with the door being open/closed then the layerchangeonweld can raise an event or the door system can remove the component when open or something